### PR TITLE
MST-151 Add the error messages handling while saving the curriculum

### PR DESCRIPTION
**Pull request by schenedx**
@edx/masters-devs-cosmonauts and @Dillon-Dumesnil Please review
We need to add this error handling so it shows clear error messages when encountering newly introduced integrity errors.
See the screenshot below

curriculumsaveerror

### CODE REVIEW

File: course_discovery/apps/course_metadata/admin.py [hunk 1]
```diff
@@ -434,6 +435,13 @@ class CurriculumAdmin(admin.ModelAdmin):
     list_display = ('uuid', 'program', 'name', 'is_active')
     inlines = (CurriculumProgramMembershipInline, CurriculumCourseMembershipInline)
 
+    def save_model(self, request, obj, form, change):
+        try:
+            super().save_model(request, obj, form, change)
+        except (IntegrityError):
+            self.save_error = True
+            logger.exception('An database integrity error occurred while saving curriculum [%s].', obj.uuid)
```
Comment (alangsto) [inline course_discovery/apps/course_metadata/admin.py]: Small typo, 'A database integrity'

**Review by alangsto — APPROVED** [review]
Small typo, otherwise LGTM

File: course_discovery/apps/course_metadata/admin.py (hunk 1 above)
Comment (schenedx) [reply course_discovery/apps/course_metadata/admin.py]: Good catch. Updated

File: course_discovery/apps/course_metadata/admin.py [hunk 2]
```diff
@@ -434,6 +435,13 @@ class CurriculumAdmin(admin.ModelAdmin):
     list_display = ('uuid', 'program', 'name', 'is_active')
     inlines = (CurriculumProgramMembershipInline, CurriculumCourseMembershipInline)
 
+    def save_model(self, request, obj, form, change):
+        try:
+            super().save_model(request, obj, form, change)
+        except (IntegrityError):
+            self.save_error = True
+            logger.exception('A database integrity error occurred while saving curriculum [%s].', obj.uuid)
```
Comment (Dillon-Dumesnil) [inline course_discovery/apps/course_metadata/admin.py]: Should this be ending on a `raise` to continue the past behavior? Assuming save_model used to raise the error.

**Review by Dillon-Dumesnil — APPROVED** [review]
Small question to clarify desired behavior, but otherwise, looks good to me.

File: course_discovery/apps/course_metadata/admin.py (hunk 2 above)
Comment (schenedx) [reply course_discovery/apps/course_metadata/admin.py]: @Dillon-Dumesnil No, we should not keep raising this error to continue the past behavior. The database integrity error is preventing the save so we should no longer save the model as it is.

**Pull request merged**